tests/test_fileref.cpp

```diff
@@ -59,6 +59,10 @@ namespace
     {
       return new Ogg::Vorbis::File(fileName);
     }
+    virtual File *createFile(IOStream* s, bool, AudioProperties::ReadStyle) const
+    {
+      return new Ogg::Vorbis::File(s);
+    }
   };
```

Comment:
```suggestion
    {
      return new Ogg::Vorbis::File(fileName);
    }
  };

  class DummyStreamResolver : public FileRef::StreamTypeResolver
  {
  public:
    virtual File *createFile(FileName, bool, AudioProperties::ReadStyle) const
    {
      return 0;
    }

    virtual File *createFileFromStream(IOStream *s, bool, AudioProperties::ReadStyle) const
    {
      return new MP4::File(s);
    }
  };
```
This will not really work to test your resolver. First you have to implement a `StreamTypeResolver` and not just add a virtual method to a `FileTypeResolver` (and the method must be `createFileFromStream()`). You should also keep the test with a file type resolver to make sure it still works with them. The other problem is that with the existing test, `createFileFromStream()` will never be called, you have to explicitly create a `FileRef` by stream, so add another test to `testFileResolver`:

```
  void testFileResolver()
  {
    {
      FileRef f(TEST_FILE_PATH_C("xing.mp3"));
      CPPUNIT_ASSERT(dynamic_cast<MPEG::File *>(f.file()) != NULL);
    }

    DummyResolver resolver;
    FileRef::addFileTypeResolver(&resolver);

    {
      FileRef f(TEST_FILE_PATH_C("xing.mp3"));
      CPPUNIT_ASSERT(dynamic_cast<Ogg::Vorbis::File *>(f.file()) != NULL);
    }

    DummyStreamResolver streamResolver;
    FileRef::addFileTypeResolver(&streamResolver);

    {
      FileStream s(TEST_FILE_PATH_C("xing.mp3"));
      FileRef f(&s);
      CPPUNIT_ASSERT(dynamic_cast<MP4::File *>(f.file()) != NULL);
    }
  }
```

Reply:
Indeed, thanks for providing the test code! Should be fixed in the new PR version